src/server/trace/viewer/traceViewer.ts

```diff
@@ -192,6 +180,30 @@ Please run 'npx playwright install' to install Playwright browsers
     await page.mainFrame().goto(internalCallMetadata(), urlPrefix + '/traceviewer/traceViewer/index.html');
     return context;
   }
+
+  private _unzipTraceFile(tracePath: string) {
+    if (!this._unzippedTraceDir.has(tracePath)) {
+      const dir = fs.mkdtempSync(path.join(os.tmpdir(), `playwright-trace`));
+      process.on('exit', () => rimraf.sync(dir));
+      const promise = extract(tracePath, { dir }).then(() => dir).catch(e => {
+        console.log(`Invalid trace file: ${tracePath}`);  // eslint-disable-line no-console
+        return '';
+      });
+      this._unzippedTraceDir.set(tracePath, promise);
+    }
+    return this._unzippedTraceDir.get(tracePath)!;
+  }
+}
+
+function queryParamToFilePath(requestUrl: string): string | undefined {
+  try {
+    const url = new URL('http://localhost' + requestUrl);
```

Comment:
I would maybe use GET parameters with URLSearchParams and have a parameter e.g. `l` for it so it's more extensible in the future.